### ISSUE TITLE
Any way to get cmake to output an Xcode project file?

### ISSUE THREAD

**Issue by JohnCoates** [oclint/oclint#382]
I'm trying to make a new rule, but it would be useful if I could work within Xcode.

I get the following error when running `cmake -G Xcode`
````
CMake Error at /Users/macbook/Dev/Clones/oclint/oclint-core/cmake/OCLintConfig.cmake:56 (FIND_PACKAGE):
  Could not find a package configuration file provided by "LLVM" with any of
  the following names:

    LLVMConfig.cmake
    llvm-config.cmake

  Add the installation prefix of "LLVM" to CMAKE_PREFIX_PATH or set
  "LLVM_DIR" to a directory containing one of the above files.  If "LLVM"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  CMakeLists.txt:9 (INCLUDE)

````

I've tried changing the LLVM_DIR in the produced CMakeCache.txt but that doesn't work either. Just wondering if there's a recommended way to do this?

**Comment by ryuichis** [comment]
Hope http://docs.oclint.org/en/stable/devel/index.html can help.

**Comment by JohnCoates** [comment]
Which part of that @ryuichis?  I can build it fine with the instructions on that page. However, when specifically trying to output an Xcode project with CMake, the command fails. I don't see anything on that page regarding this.

**Comment by ryuichis** [comment]
@JohnCoates hey, sorry, I didn't catch your point earlier.

Frankly speaking, I am not aware of people who work on oclint project entirely inside Xcode. I know people who just use it as a syntax highlighter but not fully depend on it.

Maybe you can work with me and try to understand if using Xcode is doable. 

Back to the initial question: could you try to call the following commands inside `oclint-scripts` folder

```
./clang co
./countly co
./googleTest co
```

before running `cmake -G Xcode`?

I am not sure if it will work, but at least, a good starting point. 

I pointed the link to the document earlier, because if you give up Xcode and want to work on your new rules by using your favorite text editor, that link should have the info you need for setting things up, and the scaffolding script to ease some of the rule generations.

**Comment by JohnCoates** [comment]
Thanks for working with me on this. Below is the result I got, and unfortunately I got the same error as above when running `cmake -G Xcode`

````
Macbook:oclint-scripts macbook$ ./clang co
Clang folder exists!
Macbook:oclint-scripts macbook$ ./countly co
./googleTest co
Countly folder exists!
Macbook:oclint-scripts macbook$ ./googleTest co
Cloning into 'googletest'...
remote: Counting objects: 7422, done.
remote: Total 7422 (delta 0), reused 0 (delta 0), pack-reused 7422
Receiving objects: 100% (7422/7422), 2.49 MiB | 2.45 MiB/s, done.
Resolving deltas: 100% (5513/5513), done.
````

Unfortunately I'm not familiar with CMake scripts, so I don't know how to point it to the right direction to look for the LLVM cmake files. I'll try and look into this.

**Comment by ryuichi-assistant** [comment]
There hasn't been any activity on this issue recently. Due to the high number of incoming GitHub notifications, we have to clean some of the old issues, as many of them have already been resolved with the latest updates.

Please make sure to update to the latest version and check if that solves the issue. Let us know if that works for you by adding a comment.

**Comment by ryuichi-assistant** [comment]
This issue will be auto-closed because there hasn't been any activity for a few months. Feel free to open a new one if you still experience this problem.

**Comment by Dis3buted** [comment]
cmake -G "Xcode" ../llvm-4.0.1.src works for me

**Comment by JohnCoates** [comment]
@Dis3buted 

Running that only built an Xcode project for llvm, and I didn't see any OCLint targets. For reference here all the commands I ran:

```
git clone https://github.com/oclint/oclint
cd oclint
svn co http://llvm.org/svn/llvm-project/llvm/tags/RELEASE_401/final/ llvm-4.0.1.src
mkdir buildXcode
cd buildXcode
cmake -G "Xcode" ../llvm-4.0.1.src
```

**Comment by JohnCoates** [comment]
Figured it out! Here are the commands I ran to get oclint-rules compiling in Xcode.
After running all of these commands you need to open the Xcode projects for oclint-core and oclint-metrics and compile those before oclint-rules will compile.

```
export SOURCE_ROOT=/path/to/oclint
export BUILD_ROOT=$SOURCE_ROOT/buildXcode

git clone https://github.com/oclint/oclint $SOURCE_ROOT

# download LLVM/clang code
svn co --quiet http://llvm.org/svn/llvm-project/llvm/tags/RELEASE_500/rc2 $SOURCE_ROOT/llvm
svn co --quiet http://llvm.org/svn/llvm-project/cfe/tags/RELEASE_500/rc2 $SOURCE_ROOT/llvm/tools/clang
svn co --quiet http://llvm.org/svn/llvm-project/compiler-rt/tags/RELEASE_500/rc2 $SOURCE_ROOT/llvm/projects/compiler-rt

# Compile LLVM/clang

mkdir -p $BUILD_ROOT/llvm
cd $BUILD_ROOT/llvm
cmake -D CMAKE_BUILD_TYPE=Release -D LLVM_BUILD_RUNTIME=ON -D CMAKE_CXX_FLAGS="-std=c++11 -stdlib=libc++ ${CMAKE_CXX_FLAGS}" -D CMAKE_INSTALL_PREFIX=$BUILD_ROOT/llvm-install $SOURCE_ROOT/llvm
make -j 8
make install

# Compile oclint-core

mkdir -p $BUILD_ROOT/oclint-core
cd $BUILD_ROOT/oclint-core
cmake -G Xcode -D OCLINT_BUILD_TYPE=Release -D CMAKE_CXX_COMPILER=$BUILD_ROOT/llvm-install/bin/clang++ -D CMAKE_C_COMPILER=$BUILD_ROOT/llvm-install/bin/clang -D LLVM_ROOT=$BUILD_ROOT/llvm-install $SOURCE_ROOT/oclint-core

# Compile oclint-metrics

mkdir -p $BUILD_ROOT/oclint-metrics
cd $BUILD_ROOT/oclint-metrics
cmake -G Xcode -D OCLINT_BUILD_TYPE=Release -D CMAKE_CXX_COMPILER=$BUILD_ROOT/llvm-install/bin/clang++ -D CMAKE_C_COMPILER=$BUILD_ROOT/llvm-install/bin/clang -D LLVM_ROOT=$BUILD_ROOT/llvm-install $SOURCE_ROOT/oclint-metrics

# Compile oclint-rules

mkdir -p $BUILD_ROOT/oclint-rules
cd $BUILD_ROOT/oclint-rules
cmake -G Xcode -D OCLINT_BUILD_TYPE=Release -D CMAKE_CXX_COMPILER=$BUILD_ROOT/llvm-install/bin/clang++ -D CMAKE_C_COMPILER=$BUILD_ROOT/llvm-install/bin/clang -D OCLINT_BUILD_DIR=$BUILD_ROOT/oclint-core -D OCLINT_SOURCE_DIR=$SOURCE_ROOT/oclint-core -D OCLINT_METRICS_SOURCE_DIR=$SOURCE_ROOT/oclint-metrics -D OCLINT_METRICS_BUILD_DIR=$BUILD_ROOT/oclint-metrics -D LLVM_ROOT=$BUILD_ROOT/llvm-install $SOURCE_ROOT/oclint-rules
```

**Comment by JohnCoates** [comment]
Thanks for the suggestions @ryuichis. I've attached a rst document
 with the instructions, how does it look?
[xcodeDevelopmentEnvironment.txt](https://github.com/oclint/oclint/files/1239947/xcodeDevelopmentEnvironment.txt)

**Comment by ryuichis** [comment]
Nice, by looking at this, I started to think it will be better to add a script to `oclint-scripts` folder that automates the entire thing. And then, your document could be simplified to just call that script. What do you think?

**Comment by JohnCoates** [comment]
Yeah that sounds like a simpler solution!

**Comment by ryuichi-assistant** [comment]
There hasn't been any activity on this issue recently. Due to the high number of incoming GitHub notifications, we have to clean some of the old issues, as many of them have already been resolved with the latest updates.

Please make sure to update to the latest version and check if that solves the issue. Let us know if that works for you by adding a comment.
